backend/de.metas.contracts/src/main/java/de/metas/contracts/bpartner/service/OrgChangeService.java

```diff
@@ -0,0 +1,488 @@
+			}
+			else
+			{
+				final BPartnerContact newContact = orgChangeRepo.createNewContact(sourceContact);
+				newContacts.add(newContact);
+
+				loggable.addLog("Contact {} was created for the destination partner {}.",
+								newContact,
+								destinationBPartnerComposite.getBpartner());
+			}
+		}
+
+		return newContacts;
+	}
+
+	private BPartnerContactType updateMatchingContactType(
+			@NonNull final DefaultContacts sourceDefaultContacts,
+			@NonNull final BPartnerContact sourceContact,
+			@NonNull final BPartnerContact matchingContact)
+	{
+		final BPartnerContactType existingContactType = sourceContact.getContactType();
+
+		final BPartnerContactType newContactType = existingContactType.deepCopy();
+
+		if (!sourceDefaultContacts.isFoundDefaultContact())
+		{
+			newContactType.setDefaultContact(matchingContact.getContactType().getIsDefaultContactOr(false));
+		}
+		if (!sourceDefaultContacts.isFoundBillToDefaultContact())
+		{
+			newContactType.setBillToDefault(matchingContact.getContactType().getIsBillToDefaultOr(false));
+		}
+		if (!sourceDefaultContacts.isFoundShipToDefaultContact())
+		{
+			newContactType.setShipToDefault(matchingContact.getContactType().getIsShipToDefaultOr(false));
+		}
+		if (!sourceDefaultContacts.isFoundSalesDefaultContact())
+		{
+			newContactType.setSalesDefault(matchingContact.getContactType().getIsSalesDefaultOr(false));
+		}
+		if (!sourceDefaultContacts.isFoundPurchaseDefaultContact())
+		{
+			newContactType.setPurchaseDefault(matchingContact.getContactType().getIsPurchaseDefaultOr(false));
+		}
+		return newContactType;
+	}
+
+	private BPartnerLocationType updateMatchingLocationType(
+			@NonNull final DefaultLocations sourceDefaultLocations,
+			@NonNull final BPartnerLocation sourceLocation,
+			@NonNull final BPartnerLocation matchingLocation)
+	{
+		final BPartnerLocationType existingLocationType = sourceLocation.getLocationType();
+
+		final BPartnerLocationType newLocationType = existingLocationType.deepCopy();
+
+		if (!sourceDefaultLocations.isFoundBillToDefaultLocation())
+		{
+			newLocationType.setBillToDefault(matchingLocation.getLocationType().getIsBillToDefaultOr(false));
+		}
+		if (!sourceDefaultLocations.isFoundShipToDefaultLocation())
+		{
+			newLocationType.setShipToDefault(matchingLocation.getLocationType().getIsShipToDefaultOr(false));
+		}
+
+		return newLocationType;
+	}
+
+	private void unmarkDefaultContactsFromDestination(@NonNull final DefaultContacts sourceDefaultContacts,
+			@NonNull final BPartnerComposite destinationBPartnerComposite)
+	{
+		final List<BPartnerContact> destinationContacts = destinationBPartnerComposite.getContacts();
+
+		if (sourceDefaultContacts.isFoundDefaultContact())
+		{
+			orgChangeRepo.unmarkDefaultContacts(destinationContacts);
+			loggable.addLog("The default contact will be the counterpart of the initial contact {}."
+									+ " Mark all the remaining contacts in the destination partner {} as non-default",
```

Comment:
```suggestion
									+ " -> Mark all the remaining contacts in the destination partner {} as non-default",
```
i use this `->` elsewhere and think it makes the log easier to read (i.e. distinguish which is the "cause" and which the "consequence")

Reply:
Changed